PolyEngine/UnitTests/Src/TransformComponentTests.cpp

```diff
@@ -0,0 +1,138 @@
+#include <catch.hpp>
+
+#include <TransformComponent.hpp>
+#include <Quaternion.hpp>
+#include <Angle.hpp>
+
+using namespace Poly;
+
+TEST_CASE("TransformComponent with no parent", "[TransformCompoent]") 
+{
+	TransformComponent tc;
+	Vector translationVector = Vector(1, 2, 3);
+	Vector scaleVector = Vector(4, 5, 6);
+	SECTION("Empty constructor")
+	{
+		for (int i = 0; i < 16; ++i)
+		{
+			REQUIRE(tc.GetLocalTransformationMatrix().Data[i] == (i % 5 == 0 ? 1 : 0));
+			REQUIRE(tc.GetGlobalTransformationMatrix().Data[i] == (i % 5 == 0 ? 1 : 0));
+		}
+	}
+
+	// basic transform setting
+	tc.SetLocalTranslation(translationVector);
+	tc.SetLocalScale(scaleVector);
+
+	SECTION("Local transformation")
+	{
+		Matrix tcl = tc.GetLocalTransformationMatrix();
```

Comment:
instead of comparing matrix fields it would be much better to simulate some real life sceniarios and compare the output witrh the prediction. For example: Create two related component and the apply transform to parent and chck if the child transfomr changed accordingly, same with rotation and scale.

You should definitely use GetGlobalRotation/Translation/Scale methods for this.

This comment applies to all unit tests in this file.